### ISSUE TITLE
Not detecting the adapter in miRNAseq

### ISSUE THREAD

**Issue by ndaniel** [OpenGene/fastp#129]
It looks like FASTP is not able to detect the adapter at all for miRNA-seq data.

For example, FASTP is not able to detect the adapter in the SE FASTQ file from https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR5087522

It was run as 'fastp -i SRR5087522.fq -o test.fq' using FASTP v0.19.6.

The first 3 input reads look like this:
```
@SRR5087522.1
TGTAACAGCAACTCCATGTGGAATGGAATTCTCGGGTGCCAAGAACTCCA
+
CCCFFFFFHHHHHJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJIJJ
@SRR5087522.2
NAGCTTATCAGACTGATGTTGACTGGAATTCTCGGGTGCCAAGGAACTCC
+
#4BDFFFFHHHHHJJJJJJJJJJJJIJJJJIJIJCBHIJJJJJJJJJJJJ
@SRR5087522.3
NCCCGGCGGCTGGGAATTCTCGGGTGCCAAGGAACTCCAGTCACCGTACG
+
#1=DDFFFHHHHHJJJJJJJJIIJFGHHIJJIEHHHACDFFFFFEDADDD
```
FASTQC shows that this fastq file contains most likely Illumina SmallRNA adapter 3', which according to FASTQC's database https://github.com/csf-ngs/fastqc/blob/master/Contaminants/contaminant_list.txt is this ATCTCGTATGCCGTCTTCTGCTTG.

**Comment by sfchen** [comment]
I will update the adapter detecting feature in next release, please help to test it then.

Thanks

**Comment by ndaniel** [comment]
Ok.

**Comment by harish0201** [comment]
Hey, just wanted to let you know that this doesn't work at times.

I think it would be better that we introduce something like an error rate for matching adapters similar to Atropos. I think they implemented that to overcome the systemic biases and 3' error rates.

**Comment by mdtorohernando** [comment]
Hi! I was trying FastP with smallRNA data and efectively, FastP does not detect these adapters.

**Comment by mdtorohernando** [comment]
Indeed, I'm trying to include a fasta file with the adapters to check... and I obtain this error: 

ERROR: the adapter <adapter_sequence> can only have bases in {A, T, C, G}, but the given sequence is: adapters_miRNAs_illumima.fasta

I attach to you the FASTA file

![image](https://github.com/OpenGene/fastp/assets/26143966/f698bad8-e6f7-477f-a860-42171a477780)
